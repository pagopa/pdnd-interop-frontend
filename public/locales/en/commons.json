{
  "productTitle": "Interoperability",
  "addBtn": "+ Aggiungi",
  "validation": {
    "mixed": {
      "required": "Il campo è obbligatorio"
    },
    "string": {
      "email": "Il formato dell'indirizzo email non è valido"
    }
  },
  "mainNav": {
    "ariaLabel": "Navigazione principale"
  }
}
