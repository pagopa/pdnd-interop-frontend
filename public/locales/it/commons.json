{
  "productTitle": "Interoperabilit√†",
  "addBtn": "+ Add",
  "validation": {
    "mixed": {
      "required": "Field required"
    },
    "string": {
      "email": "Invalid email format"
    }
  }
}
