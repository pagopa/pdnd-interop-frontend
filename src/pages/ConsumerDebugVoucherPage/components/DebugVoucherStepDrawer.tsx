import { Chip, List, ListItem, Stack, Typography } from '@mui/material'
import React from 'react'
import { useDebugVoucherContext } from '../DebugVoucherContext'
import { useTranslation } from 'react-i18next'
import { useGetDebugVoucherResultChipProps } from '../hooks/useGetDebugVoucherResultChipProps'

const DebugVoucherStepDrawer: React.FC = () => {
  const { t } = useTranslation('voucher', { keyPrefix: 'consumerDebugVoucher.result' })
  const { debugVoucherStepDrawer } = useDebugVoucherContext()

  const selectedStep = debugVoucherStepDrawer.selectedStep

  const chipProps = useGetDebugVoucherResultChipProps(selectedStep?.[1])

  return (
    <Stack spacing={2}>
      <Stack direction="row" justifyContent="space-between" alignItems="center">
        <Typography variant="body2">{t('stepDrawer.stepResultLabel')}</Typography>
        {chipProps && <Chip size="small" label={chipProps.label} color={chipProps.color} />}
      </Stack>

      <List sx={{ listStyleType: 'disc', pl: 4 }}>
        {selectedStep?.[1].failures.map((failure, index) => (
          <ListItem key={index} sx={{ display: 'list-item', px: 0 }}>
            <Typography variant="body2">
              {t(`errors.${failure.code}` as unknown as TemplateStringsArray, failure.reason)}
            </Typography>
          </ListItem>
        ))}
      </List>
    </Stack>
  )
}

export default DebugVoucherStepDrawer
