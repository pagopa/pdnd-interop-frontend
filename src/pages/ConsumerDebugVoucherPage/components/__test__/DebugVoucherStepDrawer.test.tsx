import {
  createMockDebugVoucherResultFailed,
  createMockDebugVoucherResultPassed,
  createMockDebugVoucherResultStep,
} from '__mocks__/data/voucher.mocks'
import React from 'react'
import { mockDebugVoucherContext } from './test.commons'
import { fireEvent, render } from '@testing-library/react'
import DebugVoucherStepDrawer from '../DebugVoucherStepDrawer'
import { vi } from 'vitest'

describe('DebugVoucherStepDrawer testing', () => {
  it('should match snapshot if isOpen is false', () => {
    mockDebugVoucherContext({
      response: createMockDebugVoucherResultPassed(),
      debugVoucherStepDrawer: {
        isOpen: false,
        selectedStep: ['clientAssertionValidation', createMockDebugVoucherResultStep()],
      },
    })

    const screen = render(<DebugVoucherStepDrawer />)

    expect(screen.queryByLabelText('closeIconBtn.aria-label')).not.toBeInTheDocument()

    expect(screen.baseElement).toMatchSnapshot()
  })

  it('should match snapshot if isOpen is true', () => {
    mockDebugVoucherContext({
      response: createMockDebugVoucherResultPassed(),
      debugVoucherStepDrawer: {
        isOpen: true,
        selectedStep: ['clientAssertionValidation', createMockDebugVoucherResultStep()],
      },
    })

    const screen = render(<DebugVoucherStepDrawer />)

    expect(screen.getByLabelText('closeIconBtn.aria-label')).toBeInTheDocument()

    expect(screen.baseElement).toMatchSnapshot()
  })

  it('should setDebugVoucherStepDrawer function be called if close icon is clicked', () => {
    const setDebugVoucherStepDrawerMockFn = vi.fn()

    mockDebugVoucherContext({
      response: createMockDebugVoucherResultPassed(),
      debugVoucherStepDrawer: {
        isOpen: true,
        selectedStep: ['clientAssertionValidation', createMockDebugVoucherResultStep()],
      },
      setDebugVoucherStepDrawer: setDebugVoucherStepDrawerMockFn,
    })

    const screen = render(<DebugVoucherStepDrawer />)

    const closeDrawerButton = screen.getByLabelText('closeIconBtn.aria-label')

    fireEvent.click(closeDrawerButton)

    expect(setDebugVoucherStepDrawerMockFn).toBeCalled()
  })

  it('should button nextStep not be rendered if clientKind is CONUSMER and selectedStep key is platformStatesVerification', () => {
    mockDebugVoucherContext({
      response: createMockDebugVoucherResultPassed({ clientKind: 'CONSUMER' }),
      debugVoucherStepDrawer: {
        isOpen: true,
        selectedStep: ['platformStatesVerification', createMockDebugVoucherResultStep()],
      },
    })

    const screen = render(<DebugVoucherStepDrawer />)

    expect(screen.queryByRole('button', { name: 'stepDrawer.nextStepBtn' })).not.toBeInTheDocument()
  })

  it('should button nextStep not be rendered if clientKind is API and selectedStep key is clientAssertionSignatureVerification', () => {
    mockDebugVoucherContext({
      response: createMockDebugVoucherResultPassed({
        clientKind: 'API',
        eservice: undefined,
        steps: {
          clientAssertionValidation: {
            result: 'PASSED',
            failures: [],
          },
          publicKeyRetrieve: {
            result: 'PASSED',
            failures: [],
          },
          clientAssertionSignatureVerification: {
            result: 'PASSED',
            failures: [],
          },
          platformStatesVerification: {
            result: 'SKIPPED',
            failures: [],
          },
        },
      }),
      debugVoucherStepDrawer: {
        isOpen: true,
        selectedStep: ['clientAssertionSignatureVerification', createMockDebugVoucherResultStep()],
      },
    })

    const screen = render(<DebugVoucherStepDrawer />)

    expect(screen.queryByRole('button', { name: 'stepDrawer.nextStepBtn' })).not.toBeInTheDocument()
  })

  it('should button nextStep not be rendered if clientKind is undefined and selectedStep key is clientAssertionSignatureVerification', () => {
    mockDebugVoucherContext({
      response: createMockDebugVoucherResultFailed({
        clientKind: undefined,
        eservice: undefined,
        steps: {
          clientAssertionValidation: {
            result: 'FAILED',
            failures: [
              {
                code: 'test code',
                reason: 'test reason',
              },
            ],
          },
          publicKeyRetrieve: {
            result: 'SKIPPED',
            failures: [],
          },
          clientAssertionSignatureVerification: {
            result: 'SKIPPED',
            failures: [],
          },
          platformStatesVerification: {
            result: 'SKIPPED',
            failures: [],
          },
        },
      }),
      debugVoucherStepDrawer: {
        isOpen: true,
        selectedStep: ['clientAssertionSignatureVerification', createMockDebugVoucherResultStep()],
      },
    })

    const screen = render(<DebugVoucherStepDrawer />)

    expect(screen.queryByRole('button', { name: 'stepDrawer.nextStepBtn' })).not.toBeInTheDocument()
  })

  it('should button nextStep be rendered if clientKind is CONUSMER and selectedStep key is not platformStatesVerification', () => {
    mockDebugVoucherContext({
      response: createMockDebugVoucherResultPassed({ clientKind: 'CONSUMER' }),
      debugVoucherStepDrawer: {
        isOpen: true,
        selectedStep: ['clientAssertionValidation', createMockDebugVoucherResultStep()],
      },
    })

    const screen = render(<DebugVoucherStepDrawer />)

    expect(screen.queryByRole('button', { name: 'stepDrawer.nextStepBtn' })).toBeInTheDocument()
  })

  it('should button nextStep be rendered if clientKind is API and selectedStep key is not clientAssertionSignatureVerification', () => {
    mockDebugVoucherContext({
      response: createMockDebugVoucherResultPassed({
        clientKind: 'API',
        eservice: undefined,
        steps: {
          clientAssertionValidation: {
            result: 'PASSED',
            failures: [],
          },
          publicKeyRetrieve: {
            result: 'PASSED',
            failures: [],
          },
          clientAssertionSignatureVerification: {
            result: 'PASSED',
            failures: [],
          },
          platformStatesVerification: {
            result: 'SKIPPED',
            failures: [],
          },
        },
      }),
      debugVoucherStepDrawer: {
        isOpen: true,
        selectedStep: ['publicKeyRetrieve', createMockDebugVoucherResultStep()],
      },
    })

    const screen = render(<DebugVoucherStepDrawer />)

    expect(screen.queryByRole('button', { name: 'stepDrawer.nextStepBtn' })).toBeInTheDocument()
  })

  it('should button nextStep be rendered if clientKind is undefined and selectedStep key is not clientAssertionSignatureVerification', () => {
    mockDebugVoucherContext({
      response: createMockDebugVoucherResultFailed({
        clientKind: undefined,
        eservice: undefined,
        steps: {
          clientAssertionValidation: {
            result: 'FAILED',
            failures: [
              {
                code: 'test code',
                reason: 'test reason',
              },
            ],
          },
          publicKeyRetrieve: {
            result: 'SKIPPED',
            failures: [],
          },
          clientAssertionSignatureVerification: {
            result: 'SKIPPED',
            failures: [],
          },
          platformStatesVerification: {
            result: 'SKIPPED',
            failures: [],
          },
        },
      }),
      debugVoucherStepDrawer: {
        isOpen: true,
        selectedStep: ['clientAssertionValidation', createMockDebugVoucherResultStep()],
      },
    })

    const screen = render(<DebugVoucherStepDrawer />)

    expect(screen.queryByRole('button', { name: 'stepDrawer.nextStepBtn' })).toBeInTheDocument()
  })

  it('should nextStep function be called on nextStepButton click', () => {
    const goToNextStepMockFn = vi.fn()
    mockDebugVoucherContext({
      response: createMockDebugVoucherResultPassed({ clientKind: 'CONSUMER' }),
      debugVoucherStepDrawer: {
        isOpen: true,
        selectedStep: ['clientAssertionValidation', createMockDebugVoucherResultStep()],
      },
      goToNextStep: goToNextStepMockFn,
    })

    const screen = render(<DebugVoucherStepDrawer />)

    const nextStepButton = screen.getByRole('button', { name: 'stepDrawer.nextStepBtn' })

    fireEvent.click(nextStepButton)

    expect(goToNextStepMockFn).toBeCalled()
  })
})
