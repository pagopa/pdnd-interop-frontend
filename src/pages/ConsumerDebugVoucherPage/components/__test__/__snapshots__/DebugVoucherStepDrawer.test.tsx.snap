// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DebugVoucherStepDrawer testing > should match snapshot if isOpen is false 1`] = `
<body>
  <div />
</body>
`;

exports[`DebugVoucherStepDrawer testing > should match snapshot if isOpen is true 1`] = `
<body
  style="padding-right: 1024px; overflow: hidden;"
>
  <div
    aria-hidden="true"
  />
  <div
    class="MuiDrawer-root MuiDrawer-modal MuiModal-root css-1u2w381-MuiModal-root-MuiDrawer-root"
    role="presentation"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root MuiModal-backdrop css-i9fmh8-MuiBackdrop-root-MuiModal-backdrop"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      data-testid="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation16 MuiDrawer-paper MuiDrawer-paperAnchorRight css-1160xiw-MuiPaper-root-MuiDrawer-paper"
      tabindex="-1"
    >
      <div
        class="MuiBox-root css-1m64onx"
      >
        <button
          aria-label="closeIconBtn.aria-label"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
            data-testid="CloseIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="css-1p2c9ak-MuiStack-root"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6 css-1y8v3j7-MuiTypography-root"
        >
          stepDrawer.title.clientAssertionValidation
        </h6>
        <p
          class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
        >
          stepDrawer.description.clientAssertionValidation
        </p>
        <div
          class="css-3y1137-MuiStack-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
          >
            stepDrawer.stepResultLabel
          </p>
          <div
            class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorError MuiChip-filledError css-lwe8th-MuiChip-root"
          >
            <span
              class="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
            >
              chipLabel.failed
            </span>
          </div>
        </div>
        <ul
          class="MuiList-root MuiList-padding css-ippg2v-MuiList-root"
        >
          <div
            class="MuiListItemText-root css-3viuk-MuiListItemText-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
            >
              errors.8001
            </p>
          </div>
          <div
            class="MuiListItemText-root css-3viuk-MuiListItemText-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
            >
              errors.8004
            </p>
          </div>
        </ul>
        <div
          class="MuiBox-root css-fxzslr"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth css-1fu7jd5-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            stepDrawer.nextStepBtn
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      data-testid="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;
